### ISSUE TITLE
[feature request] CLI should catch "getsockopt: connection refused" and give clearer error message

### ISSUE THREAD

**Issue by beckettsean** [influxdb/influxdb#4463]
If the `influxd` process is not running or accessible, the `influx` CLI gives cryptic error messages that don't point users to the actual issue:

```
$ influx
Connected to http://localhost:8086 version 
InfluxDB shell 0.9.4.1
> show databases
ERR: Get http://localhost:8086/query?db=&q=show+databases: dial tcp [::1]:8086: getsockopt: connection refused
>
```

Let's catch that error in the CLI and spit out something like:

```
$ influx
Connected to http://localhost:8086 version 
InfluxDB shell 0.9.4.1
> show databases
ERR: Get http://localhost:8086/query?db=&q=show+databases: dial tcp [::1]:8086: getsockopt: connection refused
No InfluxDB process was found at the expected URL. Please check your connection settings and verify the influxd process is running
>
```

**Comment by otoolep** [comment]
The CLI also spits out this message on start up:

`Connected to http://localhost:8086 version`

even though it is not connected. This is quite confusing.

**Comment by otoolep** [comment]
Fixed by #4465
